desc/optimize/optimizer.py

```diff
@@ -405,24 +395,27 @@ def callback(x_reduced):
                     allf.append(fx)
                     dx_norm = None
                     x_norm = jnp.linalg.norm(x_reduced)
-                    if verbose > 2:
-                        print_iteration_nonlinear(
-                            len(allx), None, fx, df, dx_norm, None
-                        )
+                    print_iteration_nonlinear(
+                        len(allx), None, fx, df, dx_norm, None
+                    )
 
             print_header_nonlinear()
             try:
-                result = scipy.optimize.minimize(
-                    compute_scalar_wrapped,
-                    x0=x0_reduced,
-                    args=(),
-                    method=self.method[len("scipy-") :],
-                    jac=grad_wrapped,
-                    hess=hess_wrapped,
-                    tol=gtol,
-                    callback=callback,
-                    options={"maxiter": maxiter, "disp": disp, **options},
-                )
+                #Sends stdout output of scipy method to logger
+                scipy_stringIO = StringIO()
+                with redirect_stdout(scipy_stringIO):
```

Comment:
will this print iterations as it goes, or just capture all of them and dump to log when it finishes?

Reply:
This is a good question; I'll mock up a test version and find out.  My assumption was that defining redirect_stdout with the try/yeild/finally syntax would handle the output on a line-by-line basis but I've not worked with StringIO type things in python before.

Reply:
Hmm, it does capture them all and dumps them to log when finished- the formatting output is more sensible than other methods I looked at, but it still counts it as one large log.  Unsure how to handle this then.

Reply:
I think I can emit a DEBUG-level log statement that labels the chunk of log scipy will generate in a way that keeps the text's intended columns, and add INFO level logs before and after labelling the LSQ message block's beginning and termination message clearly.